File: look-and-say/look-and-say-java/src/main/java/pl/frognar/_05/Sequence.java
package pl.frognar._05;

public class Sequence {
    public static String lookAndSay(String input) {
        String result = "";
        int count = 1;
        if (input.length() == 2) {
            if (input.charAt(0) == input.charAt(1)) {
                count++;
                result += "%s%s".formatted(count, input.charAt(0));
                return result;
            }
            else {
                result += "%s%s".formatted(count, input.charAt(0));
                result += "%s%s".formatted(count, input.charAt(1));
                return result;
            }
        }

        result += "%s%s".formatted(count, input.charAt(input.length() - 1));
        return result;
    }
}

File: look-and-say/look-and-say-py/02/sequence.py
def look_and_say(input_value: str) -> str:
    count = 1
    result = ''
    if len(input_value) == 2:
        if input_value[0] == input_value[1]:
            count += 1
            result += f'{count}{input_value[0]}'
            return result
        result += f'{count}{input_value[0]}'
        result += f'{count}{input_value[1]}'
        return result
    result += f'{count}{input_value}'
    return result
